### ISSUE TITLE
kbfsfuse with high RES, CPU

### ISSUE THREAD

**Issue by coreymwamba** [keybase/client#15288]
my log id: d2bd1e68cf05f76e2eea491c

This has been reported before but closed -- so opening a new issue.

**Comment by strib** [comment]
Hi @coreymwamba.  For some reason the arch log don't include an exact version, so it's hard for me to tell how up-to-date your release is.  If it's a bit out-data, it might be worth updating as we fixed some memory leak bugs.

As for the cpu usage, kbfs does use cpu when files are accessed (since it's implemented in user space), and from the logs it looks like something on your system was indeed constantly reading files.  We definitely need to put work in to reduce CPU usage where possible, but for now I think it's expected while you're actively accessing the file system.

**Comment by heronhaye** [comment]
@coreymwamba, are you perhaps using the Keybase package in Arch community rather than `keybase-bin` from the AUR? `keybase-bin` is packaged by us and gets more frequent updates, so you could try that. Note that `keybase-bin` includes keybase, kbfs, and the gui.

**Comment by coreymwamba** [comment]
@heronhaye yes, I'm using the one from [community]. On both machines, they are mounted in `~/keybase`. 

I will switch over to the AUR package first; then see what happens when I change the mountpoints.

**Comment by heronhaye** [comment]
Cool. The default is `$XDG_RUNTIME_DIR/keybase/kbfs/`, if that environment variable is available. On my machine, it evaluates to `/run/user/1000/keybase/kbfs`.

**Comment by coreymwamba** [comment]
The package didn't set it to `~/keybase`, I did... but I've tried to change it back to the default and it won't create the folder. Either way, switching the package seems to have helped. Thanks!

**Comment by heronhaye** [comment]
Currently you have to make the directory yourself (but this might change soon). So,

```bash
# of course change if your runtime directory is somewhere else
keybase config set mountdir /run/user/1000/keybase/kbfs
mkdir -p "$(keybase config get -d -b mountdir)"
run_keybase
```

And it should be available in the new directory. Regardless, with `keybase-bin`, we package the redirector so your files will also be available at `/keybase` anyway.

Glad to hear it's gotten better. Let us know if it spikes up again.
